site/role/manifests/ssh.pp

```diff
@@ -0,0 +1,71 @@
+class role::ssh (
+  Hash	$sshkeys    = {},
+  Variant[Boolean,String]    $ensure     = 'present',
+  Enum['psick']              $module     = 'psick',
+) {
+	
+	$sshkeys.each |$k,$v| { 
+		role::ssh::ssh_keys { $k:
+			*	=> $v,
+		}
+	}
+
+}
+
+define role::ssh::ssh_keys (
+
+  Variant[Boolean,String]    $ensure     = 'present',
+  Enum['psick']              $module     = 'psick',
+
+  Optional[String] $ssh_private_key_content = undef,
+  Optional[String] $ssh_public_key_content  = undef,
+  Optional[String] $ssh_private_key_source  = undef,
+  Optional[String] $ssh_public_key_source   = undef,
+  Optional[String] $ssh_knownhost_content   = undef,
+  Optional[String] $ssh_knownhost_source    = undef,
+
+  Boolean $ssh_keys_generate                = false,
+  ## String $home_dir     'c:\\Users\\Administrator', - For Jenkins slave @Windows
+  String $home_dir                          = '/var/lib/jenkins',
```

Comment:
I'd like to see if we can do it with existing resources (with a tweak here or there). @alvagante Can you come up with an action plan? Yes, we'd like it soon, but a day here or there is worth it to do it properly.

Reply:
The psick::users class can iterate over the  psick::users::managed define https://github.com/example42/puppet-psick/blob/jenkins/manifests/users/managed.pp which already allows, for each user to manage via params, configurable on Hiera, ssh keys and known hosts source (but not the content, at the moment, but that's easy to add).
I have to test it on Windows though, as I think it might need some tweaks there.
Let me know if to implement the above things and if that's actually all that you need.

Reply:
@alvagante Green light from me. When it's ready, let us know and we'll finally rebase our psick on upstream. Sane? 
@shapi78 I think you should rebase this branch to drop the ssh key commit, then at least we can get it into master. Sane?

Reply:
@alvagante To clarify, this will let us put two different ssh keys in the homedir of the jenkins user on our Windows Jenkins slave? So that we can use one for the `web` repo and the other for the `frontend` repo?

Reply:
@antgel you need 2 different keys in (4) different files I suppose.
Currently, for a given user, psick::users::managed just manages id_rsa and id_rsa.pub but maybe at this point it's better just to create a generic define like psick::ssh::key which just manages the keys and not the user (so somehow more similar to the current one in role).


Anyway, since the need is very limited and confined (keys for web and frontend repo for jenkins user on windows) you could manage the relevant files resource directly in role::jenkins_slave (renamed or to to profile) without the extra role::ssh stuff

Reply:
Fine. @shapi78 Please fix the ssh commit as per @alvagante and my recommendations. Indeed, why don't we just put this resource inside the existing role::jenkins_slave?